Build failure: Conan 2.0 (`nixos-unstable`)

### Steps To Reproduce

Steps to reproduce the behavior:
1. Use the following `flake.nix` to create a shell with `conan`

```nix
{
	inputs = {
		nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
		flake-utils.url = "github:numtide/flake-utils";
	};

	outputs = { self, nixpkgs, flake-utils }:
		flake-utils.lib.eachDefaultSystem (system:
		let
			pkgs = import nixpkgs { inherit system; };

			build_libs = with pkgs; [
				conan
			];
		in
		{

			devShells.default = pkgs.mkShell {
				packages = build_libs;
			};
		});
}
```

The `flake.lock` revision for nixpkgs is c8018361fa1d1650ee8d4b96294783cf564e8a7f

### Build log

https://gist.github.com/JonBoyleCoding/5e8361d3af042b508cdbd286fc3e5185

```console
       > =========================== short test summary info ============================
       > FAILED conans/test/integration/command_v2/list_test.py::TestListPrefs::test_list_pkg_ids[True] - AssertionError: assert 'default\n  z...os: Windows\n' == 'default\n  z...  ...
       > FAILED conans/test/integration/command_v2/list_test.py::TestListPrefs::test_list_pkg_ids_all_rrevs[True] - AssertionError: assert 'default\n  z...os: Windows\n' == 'default\n  z...  ...
       > FAILED conans/test/integration/command_v2/list_test.py::TestListPrefs::test_list_latest_prevs[True] - AssertionError: assert 'default\n  z...os: Windows\n' == 'default\n  z...  ...
       > FAILED conans/test/integration/command_v2/list_test.py::TestListPrefs::test_list_all_prevs[True] - AssertionError: assert 'default\n  z...os: Windows\n' == 'default\n  z...  ...
```

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@HaoZeke

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.108, NixOS, 22.11 (Raccoon), 22.11.3889.f5364316e31`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11"`
 - channels(jonathan): `"home-manager-22.11.tar.gz"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/tools/build-managers/conan/default.nix
+++ b/pkgs/development/tools/build-managers/conan/default.nix
@@ -53,6 +53,8 @@ python3.pkgs.buildPythonApplication rec {
     webtest
   ]);
 
+  __darwinAllowLocalNetworking = true;
+
   pythonImportsCheck = [
     "conan"
   ];
@@ -65,6 +67,17 @@ python3.pkgs.buildPythonApplication rec {
   disabledTests = [
     # Tests require network access
     "TestFTP"
+  ] ++ lib.optionals stdenv.isDarwin [
+    # Rejects paths containing nix
+    "test_conditional_os"
+    # Requires Apple Clang
+    "test_detect_default_compilers"
+    "test_detect_default_in_mac_os_using_gcc_as_default"
+    # Incompatible with darwin.xattr and xcbuild from nixpkgs
+    "test_dot_files"
+    "test_xcrun"
+    "test_xcrun_in_required_by_tool_requires"
+    "test_xcrun_in_tool_requires"
   ];
 
   disabledTestPaths = [

--- a/pkgs/development/tools/build-managers/conan/default.nix
+++ b/pkgs/development/tools/build-managers/conan/default.nix
@@ -9,14 +9,14 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "conan";
-  version = "2.0.0";
+  version = "2.0.5";
   format = "setuptools";
 
   src = fetchFromGitHub {
     owner = "conan-io";
     repo = "conan";
     rev = "refs/tags/${version}";
-    hash = "sha256-yx/MO5QAVKnGraQXJitXxaZooLtBqa+L04s73DwiE14=";
+    hash = "sha256-+ohUOQ9WBER/X0TDklf/qZCm9LhM1I1QRmED4FnkweM=";
   };
 
   propagatedBuildInputs = with python3.pkgs; [
@@ -71,13 +71,17 @@ python3.pkgs.buildPythonApplication rec {
     # Requires cmake, meson, autotools, apt-get, etc.
     "conans/test/functional/command/new_test.py"
     "conans/test/functional/command/test_install_deploy.py"
+    "conans/test/functional/graph/test_transitive_build_scripts.py"
+    "conans/test/functional/layout/test_editable_cmake_components.py"
     "conans/test/functional/layout/test_editable_cmake.py"
     "conans/test/functional/layout/test_in_subfolder.py"
     "conans/test/functional/layout/test_source_folder.py"
     "conans/test/functional/toolchains/"
     "conans/test/functional/tools_versions_test.py"
+    "conans/test/functional/tools/scm/test_git.py"
     "conans/test/functional/tools/system/package_manager_test.py"
     "conans/test/functional/util/test_cmd_args_to_string.py"
+    "conans/test/integration/command_v2/list_test.py"
     "conans/test/unittests/tools/env/test_env_files.py"
   ];
 

